sheeprl/utils/mlflow.py

```diff
@@ -0,0 +1,427 @@
+"""Thank you @lorenzomammana: https://github.com/orobix/quadra/blob/main/quadra/utils/model_manager.py"""
+
+from __future__ import annotations
+
+import getpass
+import os
+import warnings
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import Any, Callable, Dict, Literal, Sequence, Set
+
+import gymnasium as gym
+import torch
+from lightning import Fabric
+from lightning.fabric.wrappers import _FabricModule
+from lightning.pytorch.loggers.mlflow import MLFlowLogger
+
+from sheeprl.utils.env import make_env
+from sheeprl.utils.imports import _IS_MLFLOW_AVAILABLE
+
+if not _IS_MLFLOW_AVAILABLE:
+    raise ModuleNotFoundError(str(_IS_MLFLOW_AVAILABLE))
+
+import mlflow  # noqa
+from mlflow.entities import Run  # noqa
+from mlflow.entities.model_registry import ModelVersion  # noqa
+from mlflow.exceptions import RestException  # noqa
+from mlflow.models.model import ModelInfo  # noqa
+from mlflow.tracking import MlflowClient  # noqa
+
+VERSION_MD_TEMPLATE = "## **Version {}**\n"
+DESCRIPTION_MD_TEMPLATE = "### Description: \n{}\n"
+
+
+class AbstractModelManager(ABC):
```

Comment:
The `AbstractModelManager` and `MlflowModelManager` classes are duplicated: in this file and in `./sheeprl/utils/model_manager.py` file.
I would keep only one copy.